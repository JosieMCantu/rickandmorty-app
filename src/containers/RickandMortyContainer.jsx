import React, { Component } from 'react';
import { findCharacters } from '../services/rickAndMortyApi';
import CharacterList from '../components/characters/CharacterList';


export default class RickandMortyContainer extends Component {
    state = {
        loading: true,
        character: []
    }

    async componentDidMount() {

        const characters = await findCharacters();
        
        this.setState({
            characters,
            loading: false,
        });
    }
    render() {

    const { loading, characters } = this.state;
    if (loading) return <h1>Loading...</h1>;

    return <CharacterList characters={characters} />
    }
}
