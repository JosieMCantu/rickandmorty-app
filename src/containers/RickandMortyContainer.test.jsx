import React from 'react';
import { render, screen } from '@testing-library/react';
import RickandMortyContainer from './RickandMortyContainer';
import { MemoryRouter } from 'react-router-dom';

describe('Rick and Morty Container', () => {
    it('renders a list of characters', async () => {

        render(<MemoryRouter><RickandMortyContainer /></MemoryRouter>);

        screen.getByText('Loading...');
        
        const ul = await screen.findByRole('list', { name: 'characters' });
        expect(ul).toMatchSnapshot();
    });
});