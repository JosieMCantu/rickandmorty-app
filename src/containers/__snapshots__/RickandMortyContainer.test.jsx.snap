// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rick and Morty Container renders a list of characters 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/1"
    >
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <p>
        1
      </p>
      <p>
        Rick Sanchez
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/2"
    >
      <img
        alt="Morty Smith"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <p>
        2
      </p>
      <p>
        Morty Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/3"
    >
      <img
        alt="Summer Smith"
        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      />
      <p>
        3
      </p>
      <p>
        Summer Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/4"
    >
      <img
        alt="Beth Smith"
        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      />
      <p>
        4
      </p>
      <p>
        Beth Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5"
    >
      <img
        alt="Jerry Smith"
        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      />
      <p>
        5
      </p>
      <p>
        Jerry Smith
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/6"
    >
      <img
        alt="Abadango Cluster Princess"
        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      />
      <p>
        6
      </p>
      <p>
        Abadango Cluster Princess
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/7"
    >
      <img
        alt="Abradolf Lincler"
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      />
      <p>
        7
      </p>
      <p>
        Abradolf Lincler
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/8"
    >
      <img
        alt="Adjudicator Rick"
        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      />
      <p>
        8
      </p>
      <p>
        Adjudicator Rick
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/9"
    >
      <img
        alt="Agency Director"
        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      />
      <p>
        9
      </p>
      <p>
        Agency Director
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/10"
    >
      <img
        alt="Alan Rails"
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
      <p>
        10
      </p>
      <p>
        Alan Rails
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/11"
    >
      <img
        alt="Albert Einstein"
        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
      />
      <p>
        11
      </p>
      <p>
        Albert Einstein
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/12"
    >
      <img
        alt="Alexander"
        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
      />
      <p>
        12
      </p>
      <p>
        Alexander
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/13"
    >
      <img
        alt="Alien Googah"
        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
      />
      <p>
        13
      </p>
      <p>
        Alien Googah
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/14"
    >
      <img
        alt="Alien Morty"
        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
      />
      <p>
        14
      </p>
      <p>
        Alien Morty
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/15"
    >
      <img
        alt="Alien Rick"
        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
      />
      <p>
        15
      </p>
      <p>
        Alien Rick
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/16"
    >
      <img
        alt="Amish Cyborg"
        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
      />
      <p>
        16
      </p>
      <p>
        Amish Cyborg
      </p>
      <p>
        Dead
      </p>
    </a>
  </li>
  <li>
    <a
      href="/17"
    >
      <img
        alt="Annie"
        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
      />
      <p>
        17
      </p>
      <p>
        Annie
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/18"
    >
      <img
        alt="Antenna Morty"
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <p>
        18
      </p>
      <p>
        Antenna Morty
      </p>
      <p>
        Alive
      </p>
    </a>
  </li>
  <li>
    <a
      href="/19"
    >
      <img
        alt="Antenna Rick"
        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
      />
      <p>
        19
      </p>
      <p>
        Antenna Rick
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
  <li>
    <a
      href="/20"
    >
      <img
        alt="Ants in my Eyes Johnson"
        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
      />
      <p>
        20
      </p>
      <p>
        Ants in my Eyes Johnson
      </p>
      <p>
        unknown
      </p>
    </a>
  </li>
</ul>
`;
